api/v1beta1/disruption_webhook.go

```diff
@@ -138,15 +142,45 @@ func (r *Disruption) ValidateCreate() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Disruption) ValidateUpdate(old runtime.Object) error {
-	logger.Debugw("validating updated disruption", "instance", r.Name, "namespace", r.Namespace)
+	logger := logger.With("instance", r.Name, "namespace", r.Namespace)
+	logger.Debugw("validating updated disruption")
+
+	var err error
+
+	oldDisruption := old.(*Disruption)
+
+	// ensure finalizer removal is only allowed if no related chaos pods exists
+	// such verification is only needed if disruption is currently in removal as controller will always add finalizer to disruptions not in removal during the reconcile loop
```

Comment:
Is this resilient to a user first removing the finalizer, then deleting the disruption? Are we guaranteed the controller will reconcile between the two as a response to the update that removes the finalizer?

Reply:
I didn't think it through enough, I rescind my worries